modules/distributor/distributor.go

```diff
@@ -603,7 +616,16 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 		traces = append(traces, r)
 	}
 
-	return keys, traces, nil
+	return keys, traces, discardedSpans, nil
+}
+
+func spanContainsAttributeTooLarge(span *v1.Span, maxAttrSize int) bool {
+	for _, attr := range span.Attributes {
+		if len(attr.Key) > maxAttrSize || attr.Value.Size() > maxAttrSize {
```

Comment:
Good call, checking the attribute name.  I'm wondering if this calculation should be against the total Key.Size() + Value.Size()?

Reply:
if key is max/2 and value is max/2, does it still cause a problem?